python/src/aac/plugins/validators/_validator_findings.py

```diff
@@ -18,9 +19,10 @@ class ValidatorFindings:
 
     findings: list[ValidatorFinding] = attrib(default=Factory(list), validator=validators.instance_of(list))
 
-    def add_findings(self, new_findings: list[ValidatorFinding]) -> None:
+    def add_findings(self, new_findings: Union[list[ValidatorFinding], "ValidatorFindings"]) -> None:
```

Comment:
Not following what making new findings a union of the list of findings and the string identifying as findings gains?

Reply:
This is just a type hint. The `Union` is Python's way of showing that the type of an object can be one of multiple things - in this case a list of `ValidatorFinding` objects or a `ValidatorFindings` object (the string is used here because in the definition of the class, Python still doesn't know what `"ValidatorFindings"` is so I think it's akin to a forward declaration in C++, or something). Basically it allows us to (using Java terms) overload the function for either type of parameter but (to my knowledge) Python doesn't allow overloading the way it's done in statically-typed languages.

The reason I'm doing this is so that we can either add findings from a list of findings (the original behavior) or get the findings from the passed in `ValidatorFindings` object to combine them (imagine overloading the `+` operator for the `ValidatorFindings` object) which allows me to extract the "add finding" part into a function and combine all the collected findings together.

Reply:
Thanks for the thorough explanation!